src/App.tsx

```diff
@@ -16,35 +17,95 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  enum SortType {
+    ALPHABET = 'alphabet',
+    LENGTH = 'length',
+    NONE = 'none',
+  }
+
+  const visibleGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    },
+  );
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const reverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
```

Comment:
The function's name 'reverse' does not start with a verb indicating what the function does. Consider renaming it to something like 'handleReverse' to improve readability and code style. [CODE STYLE]